gc.c

```diff
@@ -7021,7 +7023,9 @@ gc_aging(rb_objspace_t *objspace, VALUE obj)
     if (!RVALUE_PAGE_WB_UNPROTECTED(page, obj)) {
         if (!RVALUE_OLD_P(obj)) {
             gc_report(3, objspace, "gc_aging: YOUNG: %s\n", obj_info(obj));
-            RVALUE_AGE_INC(objspace, obj);
+            if (is_full_marking(objspace) || !gc_params.aging_factor || (objspace->profile.minor_gc_count % gc_params.aging_factor)) {
```

Comment:
We should also change `RVALUE_OLD_AGE` to scale with `aging_factor` so it doesn't trigger major GC so often. I think it defeats the purpose to trigger major GC as frequently as before.

Reply:
Hum, we can't change that define directly as it's use to set the age on objects, but I assume you mean this if:

```c
                if (objspace->profile.count - objspace->rgengc.last_major_gc < RVALUE_OLD_AGE) {
                    full_marking = TRUE;
```

In which case yeah, that should take the factor that into account.

Reply:
Yeah we can change the places that use `RVALUE_OLD_AGE` instead too.

Reply:
I think there is only two places where it may make sense, and for the `grow_heap` one I'm unsure. I applied the change to both though.